### PR TITLE
Update emission factors for Europe (2023)

### PR DESCRIPTION
**Pull request by madsnedergaard**
## Description

**Updates the emission factors for EU zones with 2023 numbers  🎉** 

New datasets from EU-ETS were released in April this year. We decided to not just use the new dataset, but also significantly revamp the code used for calculating emission factors and utilise the great work of @w-flo on mapping EU power plants 🙌

> [!NOTE]  
> See [our Wiki page](https://github.com/electricitymaps/electricitymaps-contrib/wiki/EU-emission-factors) for a deeper explanation of the methodology behind this.
> The code used to calculating these numbers is available in [this notebook](https://colab.research.google.com/drive/1IkDXX4p1xQIuY3WicL94TUkRcZQnS9c7?usp=sharing) 

_This PR replaces #7068 due to significant changes. Closes issue #6581_

### Calculation changes

While the core methodology is the same, the following changes have been implemented to the notebook used for calculation:

- Data is calculated using latest public datasets (for 2021, 2022 and 2023)
- We now use [an improved list of power plants](https://github.com/w-flo/eu-emission-factors) that means more power plants are included in the calculations
- 16 power plants with missing emission data in the EU-ETS dataset have been manually filled in
- 10 power plants with mixed production modes (mostly thermal plants converted or largely co-fired with biomass) have been excluded from the dataset (to avoid representing a specific mode incorrectly)
- Plants with incorrect production modes (most likely because the units have been converted at some point) have been fixed
- Uses updated [heat allocation numbers](https://docs.google.com/spreadsheets/d/1y27Ro5T1FeGoV9Bf8pPF52Tw4U0MX08zpJbtk9Gz7b8/edit?gid=0#gid=0) (following same methodology as this paper: https://zenodo.org/records/7316186)
- General improvements to the way we match entries between the datasets to increase matching between datasets

### Significant data changes

We have detailed all significant changes in comments below, as well as on this website: https://electricitymaps.notion.site/Changelog-entry-EU-Emission-Factors-Update-3faa5fb63b214803a5a6b5d05fed8628?pvs=4

---

### Comparison with github.com/w-flo/eu-emission-factors

In order to better compare numbers between our results and the ones by @w-flo, I have made this new notebook that loads two plant-level CSVs and compares them. The majority of the plant-level differences in emission factors are due to the different methodologies around heat allocations.

https://colab.research.google.com/drive/14MR8YhEC4bb6TvJlMGYHhEo7MC07fQ9z?usp=sharing

### Updating 2021 and 2022 emission factors

As a follow-up to this, we will also be updating 2022 and 2021 numbers based on the new calculations and power plant mapping once this PR has been merged.

### CODE REVIEW

File: config/zones/CZ.yaml [hunk 1]
```diff
@@ -100,13 +100,19 @@ emissionFactors:
...
     gas:
       - datetime: '2021-01-01'
         source: EU-ETS, ENTSO-E 2021
         value: 419.8221756
       - datetime: '2022-01-01'
         source: EU-ETS, ENTSO-E 2022
         value: 412.38
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023
+        value: 364.25
```
Comment (madsnedergaard) [inline config/zones/CZ.yaml:115]: Improved mapping of power plants (for 2022 we were unable to match any, and therefore the value is an average for Europe).

File: config/zones/DK.yaml [hunk 2]
```diff
@@ -44,13 +45,19 @@ emissionFactors:
...
     gas:
       - datetime: '2021-01-01'
         source: EU-ETS, ENTSO-E 2021
         value: 336.9719859
       - datetime: '2022-01-01'
         source: EU-ETS, ENTSO-E 2022
         value: 336.97
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023
+        value: 299.26
```
Comment (madsnedergaard) [inline config/zones/DK.yaml:60]: The number for 2022 is incorrectly too high, as the gas power plant Silkeborgværket had previously been excluded.

File: config/zones/FI.yaml [hunk 3]
```diff
@@ -182,13 +191,19 @@ emissionFactors:
...
     gas:
       - datetime: '2021-01-01'
         source: EU-ETS, ENTSO-E 2021; IPCC 2014
         value: 415.9885661
       - datetime: '2022-01-01'
         source: EU-ETS, ENTSO-E 2022; IPCC 2014
         value: 415.99
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023
+        value: 596.61
```
Comment (madsnedergaard) [inline config/zones/FI.yaml:206]: The gas power plant _Vuosaaren B-voimalaitos_ is causing the significant increase in emission factor. Emissions have increased from 87,269 in 2022 to 120,283 in 2023, while ETS allocations have decreased from 85,191 to 31,848.

File: config/zones/GB-NIR.yaml [hunk 4]
```diff
@@ -49,6 +49,10 @@ emissionFactors:
       - datetime: '2021-01-01'
```
Comment (madsnedergaard) [inline config/zones/GB-NIR.yaml:49]: As Nortern Ireland (GB-NIR) and Ireland (IE) is connected to the same grid, we're now using the same emission factor for both zones.

File: config/zones/FR.yaml [hunk 5]
```diff
@@ -161,6 +167,9 @@ emissionFactors:
       - datetime: '2022-01-01'
         source: EU-ETS, ENTSO-E 2022
         value: 755.44
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023
+        value: 657.06
```
Comment (madsnedergaard) [inline config/zones/FR.yaml:172]: Decrease due to improved power plant mapping

File: config/zones/SE.yaml [hunk 6]
```diff
@@ -223,6 +233,9 @@ emissionFactors:
       - datetime: '2022-01-01'
         source: EU-ETS, ENTSO-E 2022; IPCC 2014
         value: 1169.95
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023
+        value: 617.49
```
Comment (madsnedergaard) [inline config/zones/SE.yaml:238]: This change is due to _Karlshamnverket_ having lowered the emissions a lot from 171,331 in 2022 to 12,044 in 2023 (92% decrease).

File: config/zones/SK.yaml [hunk 7]
```diff
@@ -55,13 +55,19 @@ emissionFactors:
...
     gas:
       - datetime: '2021-01-01'
         source: EU-ETS, ENTSO-E 2021
         value: 347.5807988
       - datetime: '2022-01-01'
         source: EU-ETS, ENTSO-E 2022
         value: 347.58
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023
+        value: 514.8
```
Comment (madsnedergaard) [inline config/zones/SK.yaml:70]: The power plant *"SPP Kompresorová stanica, Trakovice"* has increased emissions from 93,258 in 2022 to 338,664 in 2023 (3.6x increase).

Generation for this plant has also increased from 254,104 MWh to 657,850 MWh in 2023.

File: config/zones/RO.yaml [hunk 8]
```diff
@@ -89,13 +89,19 @@ emissionFactors:
       - datetime: '2022-01-01'
         source: EU-ETS, ENTSO-E 2022
         value: 1085.6
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023
+        value: 1224.26
```
Comment (madsnedergaard) [inline config/zones/RO.yaml:94]: Increased production from several of the highest emitting coal power plants is driving up the average here.

File: config/zones/SK.yaml [hunk 9]
```diff
@@ -55,13 +55,19 @@ emissionFactors:
       - datetime: '2022-01-01'
         source: EU-ETS, ENTSO-E 2022
         value: 1300.76
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023
+        value: 764.9
```
Comment (madsnedergaard) [inline config/zones/SK.yaml:60]: Nováky power plant had an emission factor of 2000+, which seems too unrealistic so it has been discarded.
The two coal power plants in Slovakia has both been closed now (in Dec 2023 and April 2024).

File: config/zones/SE.yaml [hunk 10]
```diff
@@ -90,6 +94,9 @@ emissionFactors:
       - datetime: '2022-01-01'
         source: EU-ETS, ENTSO-E 2022
         value: 925.95
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023
+        value: 373.49
```
Comment (madsnedergaard) [inline config/zones/SE.yaml:99]: This change is due to Karlshamnverket having lowered the emissions a lot from 171,331 in 2022 to 12,044 in 2023 (92% decrease).

File: config/zones/GR.yaml [hunk 11]
```diff
@@ -171,13 +177,19 @@ emissionFactors:
         source: EU-ETS, ENTSO-E 2022; Oberschelp, Christopher, et al. "Global emission
           hotspots of coal power generation."
         value: 1537.247562
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023; IPCC 2014
+        value: 1278.46
```
Comment (madsnedergaard) [inline config/zones/GR.yaml:182]: Emissions for the power plant _ΑΤΜΟΗΛΕΚΤΡΙΚΟΣ ΣΤΑΘΜΟΣ ΑΗΣ ΑΓΙΟΥ ΔΗΜΗΤΡΙΟΥ_ has been reduced drastically (5,867,033 in 2022 and 3,349,960 in 2023) while generation has increased from 3,784,599 to 4,026,773 MWh in 2023. Mapping of power plants is also improved.

File: config/zones/FI.yaml [hunk 12]
```diff
@@ -182,13 +191,19 @@ emissionFactors:
...
     gas:
       - datetime: '2021-01-01'
         source: EU-ETS, ENTSO-E 2021; IPCC 2014
         value: 415.9885661
       - datetime: '2022-01-01'
         source: EU-ETS, ENTSO-E 2022; IPCC 2014
         value: 415.99
+      - datetime: '2023-01-01'
+        source: EU-ETS, ENTSO-E 2023; IPCC 2014
+        value: 596.61
```
Comment (madsnedergaard) [inline config/zones/FI.yaml:206]: The gas power plant _Vuosaaren B-voimalaitos_ is causing the significant increase in emission factor. Emissions have increased from 87,269 in 2022 to 120,283 in 2023, while ETS allocations have decreased from 85,191 to 31,848.

File: EMISSION_FACTORS_SOURCES.md
Comment (VIKTORVAV99) [inline EMISSION_FACTORS_SOURCES.md:1]: You might want to remove/modify the script that generates this file so it's not overwritten in the future.

Comment (madsnedergaard) [reply EMISSION_FACTORS_SOURCES.md:1]: Good point, do you agree on removing the content like I did or do you think it's worth keeping it? :)

Comment (VIKTORVAV99) [reply EMISSION_FACTORS_SOURCES.md:1]: It has not been updated in years and nobody has mentioned it so I'd just remove it like you did.

**Review by FelixDQ — APPROVED** [review]
LGTM! Thanks a lot for the work on updating the emission factors! This is very thorough

**Pull request merged**